emma-language/src/main/scala/org/emmalanguage/compiler/lang/backend/TranslateToDataflows.scala

```diff
@@ -53,46 +53,64 @@ private[backend] trait TranslateToDataflows extends Common {
         case Core.Lang.ValDef(sym, _, _) if highFuns contains sym => true
       }(Monoids.disj)
 
-      // Change DataBag sources to the corresponding target backend sources (but only in first-order context)
-      val ctorsChanged = withHighContext.transformWith {
-        case Attr.inh(Core.Lang.DefCall(Some(api.ModuleRef(API.bagModuleSymbol)), method, targs, argss@_*), false :: _)
-          if API.sourceOps(method) => {
-          val targetMethodDecl = backendSymbol.info.decl(method.name)
-          assert(targetMethodDecl.alternatives.size == 1, s"Target method `$targetMethodDecl` shouldn't be overloaded.")
-          val targetMethod = targetMethodDecl.asMethod
-          Core.Lang.DefCall(Some(api.ModuleRef(backendSymbol)))(targetMethod, targs: _*)(argss: _*)
-        }
-      }(disambiguatedTree).tree
+      /**
+       * This is for changing static method calls to their corresponding backend calls.
+       * @param moduleSymbol The symbol of the module of which the methods are members of.
+       * @param methods The set of methods that should be affected.
+       * @return A transformation which changes these method calls to backend calls. The backend methods
+       *         should be defined in the module `backendSymbol`, and they should not have any overloads (this is
+       *         because we look up the corresponding method just by name; adding overload resolution would be
+       *         possible but complicated because of the additional implicit parameters for the backend contexts).
+       */
+      def changeStaticCalls(moduleSymbol: u.ModuleSymbol, methods: Set[u.MethodSymbol]): u.Tree => u.Tree =
+        withHighContext.transformWith {
+          case Attr.inh(Core.Lang.DefCall(Some(api.ModuleRef(`moduleSymbol`)), method, targs, argss@_*), false :: _)
+            if methods(method) => {
+            val targetMethodDecl = backendSymbol.info.decl(method.name)
+            assert(targetMethodDecl.alternatives.size == 1,
```

Comment:
That's not enough, because we would also need to manually add the backend context implicit parameter.